### PR TITLE
Fix Delete oauth token API method

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
Rework the invalidate token oauth API method to send revoke token request to the git provider. After this request Che authorisation is unsynchronised so user is asked to re apply the authentication on new factory create step.

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/21890

### How to test this PR?

1. Deploy Che with `quay.io/ivinokur/che-server:next` image.
2. Apply [GitHub oAuth flow](https://www.eclipse.org/che/docs/stable/administration-guide/configuring-oauth-2-for-github/).
3. Start a factory from a GitHub repo: authorisation request is shown.
4. Go to <che-url>/swagger and find the **DELETE** `/oauth/token` method.
5. Input `github` and click `Execute`.
6. Delete current workspace and start a new factory from the GitHub repo.
See: authorisation page is shown.
 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by olexii4** [comment]
@vinokurig I can't check your PR  with ```quay.io/ivinokur/che-server:next``` image.
[logs-from-che-in-che-b5fd54c89-wz6tp.log](https://github.com/eclipse-che/che-server/files/10324799/logs-from-che-in-che-b5fd54c89-wz6tp.log)

**Comment by vinokurig** [comment]
@olexii4 
> I can't check your PR with quay.io/ivinokur/che-server:next image.

Fixed

**Review by olexii4 — APPROVED** [review]
I have tested it with Kubernetes infrastructure.
Знімок екрана 2022-12-30 о 17 55 03

File: wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java [hunk 1]
```diff
@@ -116,4 +118,8 @@ public OAuthToken getToken(String userId) throws IOException {
     }
     return token;
   }
+
+  public String getEndpointUrl() {
+    return trimEnd(gitlabEndpoint, '/');
```
Comment (tolusha) [inline wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java:123]: Why not to trim string at line 47 ?

File: wsmaster/che-core-api-auth-openshift/src/main/java/org/eclipse/che/security/oauth/OpenShiftOAuthAuthenticator.java [hunk 2]
```diff
@@ -91,4 +94,9 @@ public OAuthToken getToken(String userId) throws IOException {
     }
     return null;
   }
+
+  @Override
+  public String getEndpointUrl() {
+    return isNullOrEmpty(oauthEndpoint) ? "" : trimEnd(oauthEndpoint, '/');
```
Comment (tolusha) [inline wsmaster/che-core-api-auth-openshift/src/main/java/org/eclipse/che/security/oauth/OpenShiftOAuthAuthenticator.java:100]: Why not to do this at line 46 ?

**Review by tolusha — APPROVED** [review]

File: wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java (hunk 1 above)
Comment (vinokurig) [reply wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java]: done

File: wsmaster/che-core-api-auth-openshift/src/main/java/org/eclipse/che/security/oauth/OpenShiftOAuthAuthenticator.java (hunk 2 above)
Comment (vinokurig) [reply wsmaster/che-core-api-auth-openshift/src/main/java/org/eclipse/che/security/oauth/OpenShiftOAuthAuthenticator.java]: done

**Pull request merged**